ktor-server/jvm/src/io/ktor/server/application/plugins/api/ServerPluginFactory.kt

```diff
@@ -23,16 +38,9 @@ public abstract class ServerPluginFactory<Configuration : Any>(
  * @return an instance of plugin
  */
 public fun <A : Pipeline<*, ApplicationCall>, ConfigurationT : Any> A.plugin(
-    plugin: ServerPluginFactory<ConfigurationT>
-): ServerPlugin<ConfigurationT> {
-    return attributes[pluginRegistryKey].getOrNull(plugin.key)
-        ?: throw MissingApplicationPluginException(plugin.key)
-}
-
-internal fun <A : Pipeline<*, ApplicationCall>> A.findServerPlugin(
-    plugin: ServerPluginFactory<*>
+    plugin: ServerPluginFactory<ConfigurationT, *>
 ): ServerPlugin<*> {
-    return attributes[pluginRegistryKey].getOrNull(plugin.key)
+    return pluginRegistry.getOrNull(plugin.key)
```

Comment:
1. Should it call `findPluginInRoute` for the case of `RoutingScopedPlugin`?
2. Could you please write tests checking this method returns different plugin instances in different subroutes?

Reply:
1. Thanks, I will
2. No, because there is only one plugin instance installed, but its configuration is changed depending on the call.